server/controllers/ReportArticleController.js

```diff
@@ -0,0 +1,147 @@
+import db from '../models';
+import response from '../helpers/response';
+
+const { Article, Report, User } = db;
+
+/**
+ *
+ *
+ * @class ReportArticleController
+ */
+class ReportArticleController {
+  /**
+  * @descripion Allow users to report articles that violate terms and agreement
+  * @static
+  * @param {*}req Express Request Object
+  * @param {*}res Express Response object
+  * @returns {object} Json response
+  * @memberof ReportArticleController
+  */
+  static async reportArticle(req, res) {
+    try {
+      const { userId } = req.user;
+      const { complaint } = req.body;
+      const { slug } = req.params;
+
+      const getArticle = await Article.findOne({
+        where: {
+          slug
+        }
+      });
+      if (!getArticle) {
+        return response(res, 404, 'failure', 'Article not found', null, null);
+      }
+      if (userId === getArticle.dataValues.userId) {
+        return response(res, 403, 'failure', 'You cannot report your own article. Do you want to Delete the article?', null, null);
```

Comment:
Please color on your choice of this response message?

Reply:
I think a 403 is correct given that 403 indicates that the client does not have access rights to that resource. Sola, can add more color though.